packages/waku/src/router/client.ts

```diff
@@ -521,10 +584,11 @@ const InnerRouter = ({
         handleScroll();
       }
       setRoute(route);
+      executeListeners('complete', route);
       refetching.current[0]?.();
       refetching.current = null;
```

Comment:
```suggestion
      refetching.current[0]?.();
      refetching.current = null;
      executeListeners('complete', route);
```

By the way, "complete" doesn't mean "finish". It means, it's done to tell React render. So, it's the "start" of rendering.